docs/inputs.md

```diff
@@ -0,0 +1,319 @@
+
+### How can I set Inputs?
+
+As installed (without specialized plugins), Chef InSpec supports five ways of setting Inputs:
+
+ * Inline in control code, using `input('input_name', value: 42)`.
+ * In profile `inspec.yml` metadata files
+ * Using the CLI option `--input-file somefile.yaml`
+ * In kitchen-inspec, using the `verifier/inputs` settings
+ * In the Audit Cookbook, using the `node[:audit][:inputs]`
+
+In addition, Chef InSpec supports Input Plugins, which can provide optional integrations to specific key-value stores.
+
+### How does Input precedence work?
+
+#### Simple Precedence
+
+Briefly,
+
+inline DSL < metadata < ( cli-input-file or kitchen-inspec or audit-cookbook )
+
+In addition, for inherited profiles,
+
+dependent profile metadata < wrapper profile metadata
+
+This lets you override input values on the command line, as well as override child profile inline values from the parent profile.
+This matches the general behavior of InSpec v3, while also making some edge cases easier to reason about.
+
+#### The Details of Input Precedence
+
+Whenever an input provider sets a value on an input, a *priority value* is assigned to the operation.  Over the life of the input, multiple assignments, with varying priority values, may occur. When the input is evaluated, the current value is determined by finding the setting event with the highest priority.
+
+Note that this approach does not rely on execution order, nor does it rely on multiple named precedence levels. Each setting operation is preserved; this allows
+
+Some input providers allow you to set a priority when you set the value.  For example, to set a priority of 50 in a metadata file, use:
+
+```yaml
+inputs:
+- name: very_important_input
+  value: 12
+  priority: 50
+```
+
+To set a priority in DSL, use:
+```ruby
+input('also_important', value: 42, priority: 45)
+```
+
+As packaged, Chef InSpec uses the following priority values:
+
+| Input Provider                         | Priority | May change priority |
+| -------------------------------------- | -------- | ------------------- |
+| Inline DSL                             | 20       |   Yes               |
+| Metadata                               | 30       |   Yes               |
+| Metadata in a wrapper cookbook         | 35       |   Yes               |
+| CLI `--input-file` option              | 40       |   No                |
+| inspec-kitchen `inputs:` section       | 40       |   No                |
+| audit cookbook `node[:audit][:inputs]` | 40       |   No                |
+
+
+### What happened to "Attributes"?
+
+When originally introduced, the Input facility was named *Attributes*. This name was problematic, as the Chef Infra Client tool uses the same word to describe its parameterization system.
+
+ * Chef attributes have a completely different and much more complex precedence system
```

Comment:
Why is there a bulleted list? Does a sentence need to be included above e.g. "Other reasons include:" ?